suite-native/device-manager/src/components/DeviceList.tsx

```diff
@@ -0,0 +1,162 @@
+            payload: { action: 'connectDeviceButton' },
+        });
+    };
+
+    //to hide/show the list with animation
+    useEffect(() => {
+        if (isVisible) {
+            const hasNotSelectedInstancelessDevices = notSelectedInstancelessDevices.length > 0;
+
+            const paddingsHeight = VERTICAL_PADDING * 2;
+            const otherDevicesHeight =
+                notSelectedInstancelessDevices.length * (ITEM_HEIGHT + ITEMS_SPACING);
+            const separatorHeight = hasNotSelectedInstancelessDevices ? SEPARATOR_HEIGHT : 0;
+            const separatorPaddding = hasNotSelectedInstancelessDevices
+                ? SEPARATOR_VERTICAL_PADDING * 2
+                : 0;
+            const radiusesHeight = MANAGER_MODAL_BOTTOM_RADIUS * 2;
+
+            const h =
+                otherDevicesHeight +
+                radiusesHeight +
+                separatorHeight +
+                separatorPaddding +
+                BUTTON_HEIGHT +
+                paddingsHeight;
+
+            opacity.value = withDelay(100, withTiming(1, { duration: 200 }));
+            height.value = withTiming(h, { duration: 300 });
+
+            setIsShown(true);
+        } else {
+            opacity.value = withTiming(0, { duration: 300 });
+            height.value = withDelay(
+                100,
+                withTiming(0, { duration: 200 }, () => {
+                    runOnJS(setIsShown)(false);
+                }),
+            );
+        }
+    }, [height, isVisible, opacity, notSelectedInstancelessDevices.length]);
+
+    const listAnimatedStyle = useAnimatedStyle(
+        () => ({
+            opacity: opacity.value,
+            height: height.value,
+        }),
+        [],
+    );
+
+    return isShown ? (
+        <Animated.View style={[listAnimatedStyle, applyStyle(listStaticStyle)]}>
+            <VStack>
+                {notSelectedInstancelessDevices.map(d => (
+                    <DeviceItem key={d.state} id={d.id} onPress={() => onSelectDevice(d)} />
+                ))}
+                {A.isEmpty(notSelectedInstancelessDevices) ? (
+                    <Box style={applyStyle(buttonWrapperStyle)}>
+                        <Button colorScheme="tertiaryElevation1" onPress={handleConnectDevice}>
+                            <Translation id="deviceManager.connectButton.first" />
+                        </Button>
+                    </Box>
+                ) : (
+                    <>
+                        <TextDivider title="generic.orSeparator" />
+                        <Box style={applyStyle(buttonWrapperStyle)}>
+                            <Button colorScheme="tertiaryElevation1" onPress={handleConnectDevice}>
+                                <Translation id="deviceManager.connectButton.another" />
+                            </Button>
+                        </Box>
+                    </>
+                )}
+            </VStack>
+        </Animated.View>
+    ) : null;
```

Comment:
nitpick: I know it's not mandatory, but I think it's much more readable to do something like `if (!isShown) return null` and then render the component.